Raven.Database/Bundles/Replication/Tasks/ReplicationTask.cs

```diff
@@ -87,10 +88,34 @@ public class ReplicationTask : IStartupTask, IDisposable
 
         private CancellationTokenSource _cts;
 
+        private Timer _propagationTimeoutTimer;
+        private ThreadLocal<bool> isThreadProcessingReplication = new ThreadLocal<bool>();
+
+        /// <summary>
+        /// Indicates that this thread is doing replication.
+        /// </summary>
+        internal ThreadLocal<bool> IsThreadProcessingReplication { get { return isThreadProcessingReplication; } set { isThreadProcessingReplication = value; } }
+
+        private bool CheckIfShouldWakeUpReplicationAndResetTimerIfNeeded()
+        {          
+            //If ThreadDoingReplication.Value is false it means that the source of the document isn't from replication controller,
+            //thus we should wake up replication right away.
+            if (!IsThreadProcessingReplication.Value)
+            {
+                _propagationTimeoutTimer.Stop();
+                return true;
+            }
+            _propagationTimeoutTimer.Start();
+            //No need to replicate right away we can wait and send a bigger batch.
+            return false;
+        }
+
         public void Execute(DocumentDatabase database)
         {
             docDb = database;
             _cts = CancellationTokenSource.CreateLinkedTokenSource(database.WorkContext.CancellationToken);
+            _propagationTimeoutTimer = new Timer() {AutoReset = false,Interval = docDb.Configuration.Replication.ReplicationPropagationDelayInSeconds*1000};
```

Comment:
I don't want to check if i need to fire the replication event...
this requires me to save a state.
The behavior of the timer is just what we need.
1.if we stop it and start it again it will run from the beginning 
2. if already started running start again is a noop